kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11584218?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11584218?buildTab=artifacts#/) on release-22.2 @ [11dd55551c651210484a1bed2c25e8a956918f37](https://github.com/cockroachdb/cockroach/commits/11dd55551c651210484a1bed2c25e8a956918f37):

Fatal error:

```
panic: priority range ID already set: old=2, new=54, first set at:

```
Stack: 

```
goroutine 51979 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc005e45678, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc005e45600, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:243 +0x58
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc005b85900, {0x8cf4eb8, 0xc006f18a50}, 0xc00325bdc0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:371 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).loadRaftMuLockedReplicaMuLocked(0xc005b85900, 0xc00325bdc0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:208 +0x98a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newReplica({0x8cf4eb8, 0xc001f3a570}, 0xc0005bf3d0?, 0x101010000000010?, 0x1?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:54 +0x185
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start.func2({0x2, {0xc0005bf3b0, 0xb, 0x10}, {0xc0005bf3c0, 0xb, 0x10}, {0xc0005bf3d0, 0x1, 0x1}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:2020 +0x273
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk.func1({{0xc008670108, 0x15, 0x78}, {{0xc008670127, 0x2d, 0x59}, {0x1780bdfaa57cd486, 0x0, 0x0}}})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1859 +0x3f5
github.com/cockroachdb/cockroach/pkg/storage.MVCCIterate({0x8cf4eb8, 0xc001f3a570}, {0x8d35450, 0xc00017c300}, {0xc0074df640, 0x9, 0x10}, {0xc0074df660, 0xb, 0x10}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:3915 +0x7e6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk({0x8cf4eb8, 0xc001f3a570}, {0x8d35450, 0xc00017c300}, 0xc003be8cb0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1869 +0x338
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc0037b0000, {0x8cf4eb8, 0xc003343770}, 0xc00517c240)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1997 +0x1014
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc004d90a80, {0x8cf4eb8, 0xc003343770}, {0x8cd19a8, _}, {_, _}, {_, _}, {0x1, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:477 +0x127a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc003ef9200, {0x8cf4e48, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1456 +0x3cd7
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0x0?, {0x8cf4e48, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1014 +0x45
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc004d08000, {0x8cf4e48, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:590 +0x67
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x8cbcfe0, 0xc004a6a580}, {0x8cbcd00, 0xc004fec340}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:332 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc001667040)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:1995 +0x477
testing.tRunner(0xc001667040, 0x6d65520)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=54, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries/write_too_old_with_put
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109871 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1 release-blocker]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31129

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11599822?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11599822?buildTab=artifacts#/) on release-22.2 @ [104d19503e192803e4969027d2cfabc0e645c3d8](https://github.com/cockroachdb/cockroach/commits/104d19503e192803e4969027d2cfabc0e645c3d8):

Fatal error:

```
panic: priority range ID already set: old=2, new=54, first set at:

```
Stack: 

```
goroutine 51713 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0027eff78, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc0027eff00, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:243 +0x58
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc006119900, {0x8cf50b8, 0xc006c73260}, 0xc009759f80)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:371 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).loadRaftMuLockedReplicaMuLocked(0xc006119900, 0xc009759f80)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:208 +0x98a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newReplica({0x8cf50b8, 0xc0019762d0}, 0xc006dee5d0?, 0x101010000000010?, 0x1?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:54 +0x185
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start.func2({0x2, {0xc006dee5b0, 0xb, 0x10}, {0xc006dee5c0, 0xb, 0x10}, {0xc006dee5d0, 0x1, 0x1}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:2020 +0x273
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk.func1({{0xc006067188, 0x15, 0x78}, {{0xc0060671a7, 0x2d, 0x59}, {0x17810d71cc1c0b75, 0x0, 0x0}}})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1859 +0x3f5
github.com/cockroachdb/cockroach/pkg/storage.MVCCIterate({0x8cf50b8, 0xc0019762d0}, {0x8d35650, 0xc0045c4000}, {0xc007b9b3c0, 0x9, 0x10}, {0xc007b9b3e0, 0xb, 0x10}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:3915 +0x7e6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk({0x8cf50b8, 0xc0019762d0}, {0x8d35650, 0xc0045c4000}, 0xc002a0acb0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1869 +0x338
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc001ae3500, {0x8cf50b8, 0xc0026e5a10}, 0xc004fa4870)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1997 +0x1014
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc004d0d500, {0x8cf50b8, 0xc0026e5a10}, {0x8cd1ba8, _}, {_, _}, {_, _}, {0x1, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:477 +0x127a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc004a07200, {0x8cf5048, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1456 +0x3cd7
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0x0?, {0x8cf5048, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1014 +0x45
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc003934000, {0x8cf5048, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:590 +0x67
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x8cbd1e0, 0xc003c302c0}, {0x8cbcf00, 0xc003712300}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:332 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc001922b60)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:1995 +0x477
testing.tRunner(0xc001922b60, 0x6d65720)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=54, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries/write_too_old_with_locking_read
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109871 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11618780?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11618780?buildTab=artifacts#/) on release-22.2 @ [eba3e91b9fe08273c611b26faa364b8d077fd5ee](https://github.com/cockroachdb/cockroach/commits/eba3e91b9fe08273c611b26faa364b8d077fd5ee):

Fatal error:

```
panic: priority range ID already set: old=2, new=54, first set at:

```
Stack: 

```
goroutine 51882 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0054b0d78, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc0054b0d00, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:243 +0x58
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc003af5900, {0x8cf56d8, 0xc00ca9de00}, 0xc0044f37a0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:371 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).loadRaftMuLockedReplicaMuLocked(0xc003af5900, 0xc0044f37a0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:208 +0x98a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newReplica({0x8cf56d8, 0xc003fb1590}, 0xc00556de00?, 0x101010000000010?, 0x1?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:54 +0x185
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start.func2({0x2, {0xc00556dde0, 0xb, 0x10}, {0xc00556ddf0, 0xb, 0x10}, {0xc00556de00, 0x1, 0x1}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:2038 +0x273
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk.func1({{0xc00705ed08, 0x15, 0x78}, {{0xc00705ed27, 0x2d, 0x59}, {0x1781ac00812a6702, 0x0, 0x0}}})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1878 +0x655
github.com/cockroachdb/cockroach/pkg/storage.MVCCIterate({0x8cf56d8, 0xc003fb1590}, {0x8d35c70, 0xc0081a8000}, {0xc00556cd40, 0x9, 0x10}, {0xc00556cd60, 0xb, 0x10}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:3915 +0x7e6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk({0x8cf56d8, 0xc003fb1590}, {0x8d35c70, 0xc0081a8000}, 0xc0024d8cb0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1888 +0x4b8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc004f22000, {0x8cf56d8, 0xc00e51a480}, 0xc0082dc240)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:2015 +0x1014
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc004d5ea80, {0x8cf56d8, 0xc00e51a480}, {0x8cd21c8, _}, {_, _}, {_, _}, {0x1, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:477 +0x127a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc004a26900, {0x8cf5668, 0xc00013c008})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1456 +0x3cd7
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0x0?, {0x8cf5668, 0xc00013c008})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1014 +0x45
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc004270b00, {0x8cf5668, 0xc00013c008})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:590 +0x67
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x8cbd800, 0xc001d72580}, {0x8cbd520, 0xc004fae300}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:332 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc001c26000)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:1995 +0x477
testing.tRunner(0xc001c26000, 0x6d65ba0)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=54, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_too_many_refreshes_in_batch_commit_with_refresh
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109871 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11629654?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11629654?buildTab=artifacts#/) on release-22.2 @ [eba3e91b9fe08273c611b26faa364b8d077fd5ee](https://github.com/cockroachdb/cockroach/commits/eba3e91b9fe08273c611b26faa364b8d077fd5ee):

Fatal error:

```
panic: priority range ID already set: old=2, new=54, first set at:

```
Stack: 

```
goroutine 51930 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc003741278, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc003741200, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:243 +0x58
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc002b54580, {0x8cf56d8, 0xc007659170}, 0xc00051a930)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:371 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).loadRaftMuLockedReplicaMuLocked(0xc002b54580, 0xc00051a930)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:208 +0x98a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newReplica({0x8cf56d8, 0xc00659c780}, 0xc005980eb0?, 0x101010000000010?, 0x1?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:54 +0x185
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start.func2({0x2, {0xc005980e90, 0xb, 0x10}, {0xc005980ea0, 0xb, 0x10}, {0xc005980eb0, 0x1, 0x1}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:2038 +0x273
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk.func1({{0xc00945b008, 0x15, 0x78}, {{0xc00945b027, 0x2d, 0x59}, {0x1781fac80d49c7c1, 0x0, 0x0}}})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1878 +0x655
github.com/cockroachdb/cockroach/pkg/storage.MVCCIterate({0x8cf56d8, 0xc00659c780}, {0x8d35c70, 0xc006cfe000}, {0xc008547e10, 0x9, 0x10}, {0xc008547e30, 0xb, 0x10}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:3915 +0x7e6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.IterateRangeDescriptorsFromDisk({0x8cf56d8, 0xc00659c780}, {0x8d35c70, 0xc006cfe000}, 0xc00484acb0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1888 +0x4b8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc002b24000, {0x8cf56d8, 0xc008503470}, 0xc0058a22d0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:2015 +0x1014
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc00192ca80, {0x8cf56d8, 0xc008503470}, {0x8cd21c8, _}, {_, _}, {_, _}, {0x1, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:477 +0x127a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc003887200, {0x8cf5668, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1456 +0x3cd7
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0x0?, {0x8cf5668, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1014 +0x45
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc0068ba000, {0x8cf5668, 0xc000084098})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:590 +0x67
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x8cbd800, 0xc004ef8840}, {0x8cbd520, 0xc004dc65c0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:332 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc001d77d40)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:1995 +0x477
testing.tRunner(0xc001d77d40, 0x6d65ba0)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=54, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_too_many_refreshes_in_batch_commit_with_refresh
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109871 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
@@ -1990,6 +1990,8 @@ func TestTxnCoordSenderRetries(t *testing.T) {
 	// operates between keys "a" and "z" and expects a single split point at "b".
 	// See the call to setupMultipleRanges below.
 	storeKnobs.DisableLoadBasedSplitting = true
+	// Similarly, disable the merge queue to avoid unexpected merges.
+	storeKnobs.DisableMergeQueue = true
 
 	var refreshSpansCondenseFilter atomic.Value
 	s, _, _ := serverutils.StartServer(t,

